incubator/nginx/nginx.libsonnet

```diff
@@ -0,0 +1,149 @@
+      },
+
+      noVhost(namespace, name, labels={app: name})::
+        base(namespace, name, labels),
+
+      hasVhost(namespace, name, labels={app: name})::
+        local volume = {
+          name: "nginx-vhost",
+          configMap: { name: name },
+        };
+        local dataMount = {
+          name: "nginx-vhost",
+          mountPath: "/bitnami/nginx/conf/vhosts",
+        };
+        base(namespace, name, labels) +
+        deployment.mixin.spec.template.spec.volumes(volume) +
+        deployment.mapContainersWithName(
+          [name],
+          function(c) c + container.volumeMounts(dataMount)
+        ),
+
+      local base(namespace, name, labels) = {
+        apiVersion: "extensions/v1beta1",
+        kind: "Deployment",
+        metadata: {
+          namespace: namespace,
+          name: name,
+          labels: { app: name },
+        },
+        spec: {
+          replicas: 1,
+          template: {
+            metadata: { labels: labels },
+            spec: {
+              containers: [{
+                name: name,
+                image: "bitnami/nginx:" + defaults.imageTag,
+                imagePullPolicy: defaults.imagePullPolicy,
+                ports: [
+                  {
+                    name: "http",
+                    containerPort: 80,
+                  },
+                  {
+                    name: "https",
+                    containerPort: 443,
+                  },
+                ],
+                livenessProbe: {
+                  httpGet: {
+                    path: "/",
+                    port: "http",
+                  },
+                  initialDelaySeconds: 30,
+                  timeoutSeconds: 5,
+                  failureThreshold: 6,
+                },
+                readinessProbe: {
+                  httpGet: {
+                    path: "/",
+                    port: "http",
+                  },
+                  initialDelaySeconds: 5,
+                  timeoutSeconds: 3,
+                  periodSeconds: 5,
+                },
+                volumeMounts: [{
+                  name: "nginx-data",
+                  mountPath: "/bitnami/nginx",
+                }]
+              }],
+              volumes: [{
+                name: "nginx-data",
+                emptyDir: {},
+              }]
+            },
+          },
+        },
+      },
+    },
+
+    service(namespace, name, selector={app: name}):: {
+      apiVersion: "v1",
+      kind: "Service",
+      metadata: {
+        namespace: namespace,
+        name: name,
+        labels: { app: name },
+      },
+      spec: {
+        type: "LoadBalancer",
+        ports: [
+          {
+            name: "http",
+            port: 80,
+            targetPort: "http",
+          },
+          {
+            name: "https",
+            port: 443,
+            targetPort: "https",
+          },
+        ],
+        selector: selector,
+      },
+    },
+
+    vhost(namespace, name):: {
```

Comment:
How about calling it something like `serverBlockConfigMap`? (nginx actually calls them server blocks instead of vhosts.)